drivers/include/fxos8700.h

```diff
@@ -0,0 +1,293 @@
+#define FXOS8700_REG_A_VECM_CNT         (0x62)
+#define FXOS8700_REG_A_VECM_INITX_MSB   (0x63)
+#define FXOS8700_REG_A_VECM_INITX_LSB   (0x64)
+#define FXOS8700_REG_A_VECM_INITY_MSB   (0x65)
+#define FXOS8700_REG_A_VECM_INITY_LSB   (0x66)
+#define FXOS8700_REG_A_VECM_INITZ_MSB   (0x67)
+#define FXOS8700_REG_A_VECM_INITZ_LSB   (0x68)
+#define FXOS8700_REG_M_VECM_CFG         (0x69)
+#define FXOS8700_REG_M_VECM_THS_MSB     (0x6A)
+#define FXOS8700_REG_M_VECM_THS_LSB     (0x6B)
+#define FXOS8700_REG_M_VECM_CNT         (0x6C)
+#define FXOS8700_REG_M_VECM_INITX_MSB   (0x6D)
+#define FXOS8700_REG_M_VECM_INITX_LSB   (0x6E)
+#define FXOS8700_REG_M_VECM_INITY_MSB   (0x6F)
+#define FXOS8700_REG_M_VECM_INITY_LSB   (0x70)
+#define FXOS8700_REG_M_VECM_INITZ_MSB   (0x71)
+#define FXOS8700_REG_M_VECM_INITZ_LSB   (0x72)
+#define FXOS8700_REG_A_FFMT_THS_X_MSB   (0x73)
+#define FXOS8700_REG_A_FFMT_THS_X_LSB   (0x74)
+#define FXOS8700_REG_A_FFMT_THS_Y_MSB   (0x75)
+#define FXOS8700_REG_A_FFMT_THS_Y_LSB   (0x76)
+#define FXOS8700_REG_A_FFMT_THS_Z_MSB   (0x77)
+#define FXOS8700_REG_A_FFMT_THS_Z_LSB   (0x78)
+/** @} */
+
+/**
+ * @brief   FXOS8700 specific return values
+ */
+enum {
+    FXOS8700_OK      = 0,     /**< everything went as expected */
+    FXOS8700_ADDRERR = -1,    /**< no FXOS8700 device found on the bus */
+    FXOS8700_NOBUS   = -2,    /**< errors while initializing the I2C bus */
+    FXOS8700_BUSERR  = -3     /**< error during I2C communication */
+};
+
+/**
+ * @brief   Parameters needed for device initialization
+ */
+typedef struct {
+    i2c_t i2c;              /**< I2C device that sensor is connected to */
+    uint8_t addr;           /**< I2C address of this particular sensor */
+} fxos8700_params_t;
+
+/**
+  * @brief   Device descriptor for a AT30TSE75x device
+  * @{
+  */
+typedef struct {
+    fxos8700_params_t p;
+    uint8_t whoami;
+} fxos8700_t;
+
+/**
+  * @brief   Individual hybrid measurement
+  * @{
+  */
+typedef struct {
+    int16_t x;
+    int16_t y;
+    int16_t z;
+} fxos8700_measurement_t;
+
+typedef union {
+    struct {
+        uint8_t  zyxow:1;     /*!< bit:      0  Start Conversion Event In          */
+        uint8_t  zow:1;       /*!< bit:      1  Synchronization Event In           */
+        uint8_t  yow:1;       /*!< bit:  2.. 3  Reserved                           */
+        uint8_t  xow:1;       /*!< bit:      0  Start Conversion Event In          */
+        uint8_t  zyxdr:1;     /*!< bit:      1  Synchronization Event In           */
+        uint8_t  zdr:1;       /*!< bit:  2.. 3  Reserved                           */
+        uint8_t  ydr:1;       /*!< bit:      0  Start Conversion Event In          */
+        uint8_t  xdr:1;       /*!< bit:      1  Synchronization Event In           */
+    } bit;                    /*!< Structure used for bit  access                  */
+    uint8_t reg;              /*!< Type      used for register access              */
+} FXOS8700_STATUS_Type;
```

Comment:
This type, and the other bit fields below, are not actually used by the implementation.
Additionally, it is not required by a compiler to pack the bit fields in the struct, so this bit representation will not be guaranteed to match the hardware register bits. It is more robust to use bit mask macros or other bit operations in the implementation. The compiler should be smart enough to optimize the accesses in a good way when one of the operands is a numeric literal or static constant.